app-antivirus/fangfrisch/fangfrisch-1.9.0.ebuild

```diff
@@ -0,0 +1,76 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{10..12} )
+
+inherit distutils-r1 readme.gentoo-r1 systemd
+
+DESCRIPTION="Update and verify unofficial Clam Anti-Virus signatures"
+HOMEPAGE="https://github.com/rseichter/fangfrisch https://pypi.org/project/fangfrisch/"
+SRC_URI="https://github.com/rseichter/fangfrisch/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+MY_CONF="/etc/${PN}.conf"
+MY_DBDIR="/var/lib/${PN}"
+DISABLE_AUTOFORMATTING=1
+DOC_CONTENTS="See https://rseichter.github.io/fangfrisch/ for the official
+documentation.
+
+### Fresh installations:
+
+Modify ${MY_CONF} according to your preferences.
+Assuming you place the database into ${MY_DBDIR}
+(recommended), execute the following commands in a root shell:
+
+mkdir -m 0770 ${MY_DBDIR}
+chgrp clamav ${MY_DBDIR}
+sudo -u clamav -- fangfrisch -c ${MY_CONF} initdb
+
+You can now enable /etc/cron.d/${PN} for periodic updates.
+
+### Updating from a previous release:
+
+Either create a fresh database or manually delete all existing
+database tables, then run the initdb command as shown above."
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND=">=dev-python/requests-2.22.0[${PYTHON_USEDEP}]
+	>=dev-python/sqlalchemy-1.3.11[${PYTHON_USEDEP}]"
+RDEPEND="${DEPEND}"
+
+distutils_enable_tests unittest
+
+python_prepare_all() {
+	sed -i -e '/SQLAlchemy/d' setup.cfg || die
+	# Mitigate build system warnings, see
+	# https://projects.gentoo.org/python/guide/qawarn.html#stray-top-level-files-in-site-packages
+	cat >>setup.cfg <<EOT
+[options.packages.find]
+exclude =
+	tests
+	tests.*
+EOT
```

Comment:
I am curious if this is something you can fix upstream...

Reply:
It is only a problem in Gentoo. Arch and Debian packages have no problems with the tests/ directory being included, and neither does PyPI. Why Gentoo is even complaining, I don't know.

Reply:
It is a problem to package the tests directory on all distros. A big problem.

I don't know or care why it may or may not end up in the final package on Arch or Debian, but maybe you should care since it's your software, and investigate why!

Reply:
I am investigating by asking you why Gentoo, and only Gentoo, is complaining about the tests being present. 🙂 Why are tests a problem, in your opinion?

Reply:
Tests must not be top level installed modules since site-packages/tests/ might then be installed by multiple projects and the files will therefore conflict between multiple packages.

`pip install` will cause multiple packages to silently overwrite each other, because pip isn't a real package manager. However, a real package manager like dpkg, pacman or portage, will fail with a "multiple packages tried to install the same file" error and the user will not be allowed to continue.

This is why portage complains when you do it. Other distros may or may not have automatic lints for this issue, but they don't exactly want installation to fail for users, so they "will* solve it by preventing the tests/ directory from being installed, if it gets reported to them.

Reply:
It is a matter of some opinion whether packages should install site-packages/myapp/tests/. I personally don't like it as they are very much not needed bloat at install time, but portage won't yell at you if you do it.

Because they don't clash with other packages if they're under the myapp/ directory rather than at the top level directly from site-packages.

Reply:
Notice how the QA warning, and the docs you link to in the ebuild, say _top-level_.

Reply:
Thank you for the explanation, that is worth considering for my future work upstream.

Reply:
FYI: I have made changes upstream, but I won't prepare a new release right away.